cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/ExecutorStateProcessor.java

```diff
@@ -0,0 +1,181 @@
+        STARTING_EXECUTION,
+        /**
+         * Indicates that there is an inter-broker partition movement task in progress.
+         */
+        INTER_BROKER_REPLICA_MOVEMENT_TASK_IN_PROGRESS,
+        /**
+         * Indicates that there is an intra-broker partition movement task in progress.
+         */
+        INTRA_BROKER_REPLICA_MOVEMENT_TASK_IN_PROGRESS,
+        /**
+         * Indicates that there is leadership partition movement task in progress.
+         */
+        LEADER_MOVEMENT_TASK_IN_PROGRESS,
+        /**
+         * Indicates the executor is in the process of stopping task execution.
+         */
+        STOPPING_EXECUTION,
+        /**
+         * Indicates the executor is preparing a task for execution.
+         */
+        INITIALIZING_PROPOSAL_EXECUTION,
+        /**
+         * Indicates the executor is generating partition movement proposals for a task for execution.
+         */
+        GENERATING_PROPOSALS_FOR_EXECUTION;
+
+        private static final List<ExecutorState> REBALANCE_EXECUTOR_STATES = List.of(
+                INTER_BROKER_REPLICA_MOVEMENT_TASK_IN_PROGRESS,
+                INTRA_BROKER_REPLICA_MOVEMENT_TASK_IN_PROGRESS,
+                LEADER_MOVEMENT_TASK_IN_PROGRESS);
+
+        /**
+         * Verifies whether the given executor state is a valid rebalancing state.
+         * If the state is not an active rebalancing state, an {@link IllegalStateException} is thrown.
+         *
+         * @param executorJson the {@link JsonNode} containing the executor state to verify;
+         *                      must have a {@code "state"} field with a valid executor state string.
+         * @throws IllegalStateException if the provided state is not a valid active rebalancing state.
+         */
+        public static void verifyRebalancingState(JsonNode executorJson) {
+            if (executorJson == null || !executorJson.has("state")) {
+                throw new IllegalStateException(
+                        String.format("Executor state: `%s` does not contain \"state\" entry", executorJson));
+            }
+
+            ExecutorState state = fromString(executorJson.get("state").asText());
+            if (!ExecutorState.REBALANCE_EXECUTOR_STATES.contains(state)) {
+                throw new IllegalStateException(
+                        String.format("Executor has not started rebalance and is currently in non-active state: '%s'. " +
+                                        "Progress estimation fields cannot be provided.",
+                                state.toString()));
+            }
+        }
+
+        private static ExecutorState fromString(String state) {
+            String normalized = state.trim();
+            return Arrays.stream(ExecutorState.values())
+                    .filter(e -> e.name().equals(normalized))
+                    .findFirst()
+                    .orElseThrow(() -> new IllegalArgumentException("Unknown ExecutorState: " + state));
+        }
```

Comment:
are we sure we need this "complicated" method to get the enum from a String returned by CC? (making a stream, filtering, ...)
I mean, CC has an OpenAPI specification and this is where the "state" is well defined in the "ExecutorState" https://github.com/linkedin/cruise-control/blob/main/cruise-control/src/main/resources/yaml/responses/executorState.yaml#L12

I think that something simpler like this would work:

```java
public static ExecutorState fromString(String state) {
    try {
        return ExecutorState.valueOf(state);
    } catch (IllegalArgumentException | NullPointerException e) {
        throw new IllegalArgumentException("Unknown ExecutorState: " + state, e);
    }
```

or am I wrong? wdyt @kyguy @tomncooper

Reply:
Yeah, if the input is constrained to these set values then you can just use `valueOf` from the super class. Although I think it might be different in Java 17 `Enum.valueOf(ExecutorState.class, state)` or similar.

Reply:
Fixed.